sparksteps/pricing.py

```diff
@@ -128,7 +150,19 @@ def get_bid_price(ec2_client, pricing_client, instance_type):
     """
     history = get_spot_price_history(ec2_client, instance_type, SPOT_PRICE_LOOKBACK)
     by_zone = price_by_zone(history)
-    zone_profile = get_zone_profile(by_zone)
+    if availability_zone:
+        # Consider only the AZ in which we expect to launch instances.
+        zone_profile = get_zone_profile({availability_zone: by_zone.get(availability_zone, [])})
+    else:
+        # Consider all AZ's.
+        zone_profile = get_zone_profile(by_zone)
+    if not zone_profile:
+        # Unable to determine the spot price because no information was available for the
+        # desired AZ.
+        logger.info(
+            "Unable to determine the spot price for %s instances in %s because no "
+            "zone information was available.", instance_type, availability_zone)
+        return get_demand_price(pricing_client, instance_type)
```

Comment:
This needs to return an `is_spot` boolean as well:
```suggestion
        return get_demand_price(pricing_client, instance_type), False
```

Also, I believe the price value will need to be rounded before return as is done below.